plugins/ImageActions/main.lua

```diff
@@ -39,11 +42,97 @@ function cb(mode)
       if image:bands()<4 then
         image = image:bandjoin(255) -- add alpha channel
       end
-      white = {240, 240, 240, 0} -- pixels are considered white, if its r,g,b values are >=240 and alpha is arbitrary
-      transparent = {0, 0, 0, 0}
+      local white = {240, 240, 240, 0} -- pixels are considered white, if its r,g,b values are >=240 and alpha is arbitrary
+      local transparent = {0, 0, 0, 0}
       image = image:more(white):bandand():ifthenelse(transparent, image)
+    elseif mode == 8 then
+      local width = image:width()
+      local height = image:height()
+      image = image:crop(image:find_trim())
+      local newWidth = image:width()
+      local newHeight = image:height()
+      maxWidth, maxHeight = maxWidth * newWidth // width, maxHeight * newHeight // height
+    end
+
+    if mode ~= 9 and mode ~=10 then
+      table.insert(imdata, {data = image:write_to_buffer(".png"), x=x, y=y, maxWidth=maxWidth, maxHeight=maxHeight})
+    else
+      local vertical = mode == 9
+      -- find largest sequence of white or transparent pixel rows/columns
+      local filter = image:colourspace("b-w"):extract_band(0):more(240)
+      local width = filter:width()
+      local height = filter:height()
+      local last = vertical and height or width
+      local isWhite = {}
+
+      for i = 0, last - 1 do
+        if vertical then
+          isWhite[i+1] = filter:extract_area(0, i, width, 1):avg() > 254
+        else
+          isWhite[i+1] = filter:extract_area(i, 0, 1, height):avg() > 254
+        end
+      end
+
+      if #isWhite == 0 then
+        word = vertical and "rows" or "columns"
```

Comment:
Agree.

I have discovered a couple of useful Lua extensions in VS Code (sumneko.lua, keyring.Lua and actboy168.lua-debug) that help greatly. Looks like there is a lot to fix in formatting and linting.

Reply:
Oh yes, I'm running `sumneko.lua` in my editor as well (ok now I realize I only used to run this, now it's `lua_ls` as sumneko doesn't seem to be supported by nvim-lspconfig anymore). It's always a nice addition.
Obviously it doesn't know about the `app` variable and it's functions by default, but I have a emmy-lua style file ([xopp_emmy.lua.txt](https://github.com/xournalpp/xournalpp/files/14530292/xopp_emmy.lua.txt)) which can be included so that the language server gets to know the library. (Tbh I'm not sure whether I've written it some time ago or if I found it somewhere (but I can't find it anymore), but it definitely isn't up to date and incomplete).

I also always get the warning with lowercase global functions but I'm always not sure how local/global works for the different plugins so I don't pay that much attention to it (but I tend to make the functions `local` as well)